streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorContextImplTest.java

```diff
@@ -0,0 +1,264 @@
+import org.apache.kafka.streams.processor.TaskId;
+import org.apache.kafka.streams.processor.internals.metrics.StreamsMetricsImpl;
+import org.apache.kafka.streams.state.KeyValueIterator;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.SessionStore;
+import org.apache.kafka.streams.state.WindowStore;
+import org.apache.kafka.streams.state.WindowStoreIterator;
+import org.apache.kafka.streams.state.internals.ThreadCache;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.easymock.EasyMock.anyLong;
+import static org.easymock.EasyMock.anyString;
+import static org.easymock.EasyMock.expect;
+import static org.easymock.EasyMock.expectLastCall;
+import static org.easymock.EasyMock.mock;
+import static org.easymock.EasyMock.replay;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class ProcessorContextImplTest {
+    private ProcessorContextImpl context;
+
+    private static final String KEY = "key";
+    private static final long VAL = 42L;
+    private static final String STORE_NAME = "underlying-store";
+
+    private boolean initExecuted;
+    private boolean closeExecuted;
+    private KeyValueIterator<String, Long> rangeIter;
+    private KeyValueIterator<String, Long> allIter;
+
+    private List<KeyValueIterator<Windowed<String>, Long>> iters = new ArrayList<>(7);
+    private WindowStoreIterator<Long> windowStoreIter;
+
+    @Before
+    public void setup() {
+        rangeIter = mock(KeyValueIterator.class);
+        allIter = mock(KeyValueIterator.class);
+        windowStoreIter = mock(WindowStoreIterator.class);
+
+        for (int i = 0; i < 7; i++)
+            iters.add(i, mock(KeyValueIterator.class));
+
+        final StreamsConfig streamsConfig = mock(StreamsConfig.class);
+        expect(streamsConfig.getString(StreamsConfig.APPLICATION_ID_CONFIG)).andReturn("add-id");
+        expect(streamsConfig.defaultValueSerde()).andReturn(Serdes.ByteArray());
+        expect(streamsConfig.defaultKeySerde()).andReturn(Serdes.ByteArray());
+        replay(streamsConfig);
+
+        final ProcessorStateManager stateManager = mock(ProcessorStateManager.class);
+
+        expect(stateManager.getGlobalStore("KeyValueStore")).andReturn(keyValueStoreMock());
+        expect(stateManager.getGlobalStore("WindowStore")).andReturn(windowStoreMock());
+        expect(stateManager.getGlobalStore("SessionStore")).andReturn(sessionStoreMock());
+
+        replay(stateManager);
+
+        context = new ProcessorContextImpl(
+            mock(TaskId.class),
+            mock(StreamTask.class),
+            streamsConfig,
+            mock(RecordCollector.class),
+            stateManager,
+            mock(StreamsMetricsImpl.class),
+            mock(ThreadCache.class)
+        );
+
+        final Set<String> stateStores = new HashSet<>();
+
+        stateStores.add("KeyValueStore");
```

Comment:
Why do we add `KeyValueStore`, but not `WindowStore`/`SessionsStore` ? If this can be any dummy, I would rename to `dummyStore` -- otherwise it might be confusing.

Reply:
This set can be replaced with empty.
Fixed